roles/openshift_aws/tasks/launch_config.yml

```diff
@@ -4,13 +4,18 @@
   when:
   - openshift_aws_ami is undefined
 
+- name: query vpc
+  ec2_vpc_net_facts:
+    region: "{{ openshift_aws_region }}"
+    filters:
+      'tag:Name': "{{ openshift_aws_vpc_name }}"
+  register: vpcout
+
 - name: fetch the security groups for launch config
   ec2_group_facts:
     filters:
-      group-name:
-      - "{{ openshift_aws_clusterid }}"  # default sg
-      - "{{ openshift_aws_clusterid }}_{{ openshift_aws_node_group_type }}"  # node type sg
-      - "{{ openshift_aws_clusterid }}_{{ openshift_aws_node_group_type }}_k8s"  # node type sg k8s
+      group-name: "{{ openshift_aws_launch_config_security_groups }}"
+      vpc-id: "{{ vpcout.vpcs[0].id }}"
```

Comment:
This line is necessary because there can be duplicate security group names in different VPCs.